File: string_compression.py
def compress(string):
	result = str()
	count = 1
	prev_s = string[0]
	string = string + " "
	for s in string[1:]:
		if s == prev_s:
			count = count + 1
		else:
			result = result + prev_s + str(count)
			count = 1
			prev_s = s
	return result if len(result) < len(string) else string

def main():
	string = "aaaaaaabca"
	ip = compress(string)
	print(ip)

if __name__ == "__main__":
	main()

File: string_compression.cpp
#include <iostream>
#include <string>

using namespace std;

string compress(string str){
	int count = 1;
	char prev_s = str[0];
	string newString = "";
	for(int i = 1; i <= str.length(); i++){
		if(prev_s == str[i]){
			count +=  1;
		} 
		else{
			newString = newString + prev_s + to_string(count);
			count = 1;
			prev_s = str[i];
		}
	}	
	if (newString.length() < str.length()){
		return newString;
	}
	else{
		return str;
	}	
}

int main(){
	string str = "aaaaaaabcs";
	cout << compress(str) << endl;

	return 0;
}
